src/coreclr/inc/slist.h

```diff
@@ -138,10 +139,10 @@ class SList
 #if 1
             // Newer compilers define offsetof to be __builtin_offsetof, which doesn't use the
             // old-school memory model trick to determine offset.
-            const UINT_PTR offset = (((UINT_PTR)&(((T *)0x1000)->*LinkPtr))-0x1000);
+            const UINT_PTR offset = (((UINT_PTR)&(((T *)0x1000)->m_Link))-0x1000);
             return (T*)__PTR(dac_cast<TADDR>(pLink) - offset);
 #else
-            return (T*)__PTR(dac_cast<TADDR>(pLink) - offsetof(T, *LinkPtr));
+            return (T*)__PTR(dac_cast<TADDR>(pLink) - offsetof(T, m_Link));
```

Comment:
We can liberally use this now based on https://grep.app/search?q=offsetof&case=true&filter[repo][0]=dotnet/runtime (and delete the `#if 1` branch).